addons/l10n_my_edi/models/account_edi_xml_ubl_my.py

```diff
@@ -142,9 +143,16 @@ def _export_invoice_vals(self, invoice):
         # The original document is mandatory; but in some specific cases it will be empty (sending a credit note for an invoice
         # managed outside Odoo/...)
         if document_type_code in ('02', '03', '04', '12', '13', '14'):
+            original_document_id = None
+            if original_document and original_document.l10n_my_edi_file_id:
+                decoded_vals = self._decode_myinvois_attachment(original_document.l10n_my_edi_file_id)
+                original_document_id = decoded_vals.get('original_document_id')
+            if not original_document_id:
+                original_document_id = original_document and original_document.ref or original_document.name or 'NA'
```

Comment:
This would only be valid for self-billed credit notes, debit notes, or refund (12, 13, 14 here).
For regular invoices/credit notes/... we never use the ref so you cannot pick it by default.

see https://github.com/odoo/odoo/blob/17.0/addons/l10n_my_edi_extended/models/account_edi_xml_ubl_my.py#L14-L25

In your code, it could work if the file is present, but it could fail in other cases.